driver-core/src/main/java/com/datastax/driver/core/policies/ClientFailureAwareRetryPolicy.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.policies;
+
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.exceptions.DriverException;
+
+/**
+ * A policy that defines a default behavior to adopt when a request returns
+ * a TimeoutException or an UnavailableException.
+ *
+ * Such policy allows to centralize the handling of query retries, allowing to
+ * minimize the need for exception catching/handling in business code.
+ */
```

Comment:
Good point. I was thinking about renaming it to `ExtendedRetryPolicy` - I don't like much the current name.